src/client/Session.cpp

```diff
@@ -34,12 +45,8 @@ bool Session::ping() {
     return conn_.ping();
 }
 
-ErrorCode Session::retryConnect() {
-    pool_->giveBack(std::move(conn_));
+void Session::retryConnect() {
     conn_ = pool_->getConnection();
```

Comment:
it maybe gets bad connection.

Reply:
If it's bad means we can't get available connection any way.